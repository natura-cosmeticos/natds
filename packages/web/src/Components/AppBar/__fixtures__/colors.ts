import { PropTypes } from '@material-ui/core'

export const colors: Record<PropTypes.Color, PropTypes.Color> = {
  default: 'default',
  inherit: 'inherit',
  primary: 'primary',
  secondary: 'secondary'
}
