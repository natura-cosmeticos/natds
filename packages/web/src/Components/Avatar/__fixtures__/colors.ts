import { AvatarColor } from '../Avatar.props'

export const colors: Record<AvatarColor, AvatarColor> = {
  primary: 'primary',
  secondary: 'secondary',
  default: 'default'
}

export default colors
