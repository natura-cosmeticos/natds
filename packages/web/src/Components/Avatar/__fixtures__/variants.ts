import { AvatarVariant } from '../Avatar.props'

/**
 * @deprecated variant is deprecated since 0.19.
 * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
 */
export const variants : Record<AvatarVariant, AvatarVariant> = {
  circle: 'circle',

  /**
   * @deprecated `rounded` variant is deprecated since 0.19.
   * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
   */
  rounded: 'rounded',

  /**
   * @deprecated `square` variant is deprecated since 0.19.
   * It will be removed at version 1.0, when all Avatar variants will be `circle` by default.
   */
  square: 'square'
}
