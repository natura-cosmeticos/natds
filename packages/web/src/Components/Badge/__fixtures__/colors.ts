import { BadgeColor } from '../Badge.props'

export const colors: Record<NonNullable<BadgeColor>, NonNullable<BadgeColor>> = {
  default: 'default',
  error: 'error',
  primary: 'primary',
  secondary: 'secondary'
}

export default colors
