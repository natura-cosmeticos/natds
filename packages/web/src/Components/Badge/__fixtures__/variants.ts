import { BadgeVariant } from '../Badge.props'

export const variants: Record<NonNullable<BadgeVariant>, NonNullable<BadgeVariant>> = {
  dot: 'dot',
  standard: 'standard'
}

export default variants
