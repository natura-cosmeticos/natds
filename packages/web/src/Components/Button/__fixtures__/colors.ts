import { ButtonColor } from '../Button.props'

export const colors : Record<ButtonColor, ButtonColor> = {
  default: 'default',
  inherit: 'inherit',
  primary: 'primary',
  secondary: 'secondary'
}
