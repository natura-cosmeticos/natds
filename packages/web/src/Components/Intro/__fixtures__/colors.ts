import { TypographyColor } from '../../Typography'

export const colors: Record<TypographyColor, TypographyColor> = {
  initial: 'initial',
  inherit: 'inherit',
  primary: 'primary',
  secondary: 'secondary',
  textPrimary: 'textPrimary',
  textSecondary: 'textSecondary',
  error: 'error'
}

export default colors
