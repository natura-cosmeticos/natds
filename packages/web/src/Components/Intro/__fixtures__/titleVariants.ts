import { TypographyVariant } from '../../Typography'

export const titleVariants : TypographyVariant[] = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6'
]

export default titleVariants
