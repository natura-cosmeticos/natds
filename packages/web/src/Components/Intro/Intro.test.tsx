// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'

import { Icon } from '../Icon'
import { Intro } from './index'
import { TypographyColor, TypographyVariant } from '../Typography'
import { Provider } from '../../Provider'
import { themes } from '../../Themes'
import { variants } from './__fixtures__/variants'
import { colors } from './__fixtures__/colors'

const TitleWithIcon = () => (
  <>
    <Icon name="filled-action-add" size="tiny" />
    title
  </>
)

describe('Intro component', () => {
  let testRenderer = {} as TestRenderer.ReactTestRenderer

  describe('when no theme is explicitly provided', () => {
    describe('and titleVariant property is undefined', () => {
      it('should match snapshot', () => {
        testRenderer = TestRenderer.create(
          // eslint-disable-next-line no-undefined
          <Intro title="Test title" titleVariant={undefined as TypographyVariant} />
        )
        expect(testRenderer).toMatchSnapshot()
      })
    })

    describe('when an invalid titleVariant property is provided', () => {
      it('should match snapshot', () => {
        testRenderer = TestRenderer.create(
          <Intro title="Test title" titleVariant={'invalid' as TypographyVariant} />
        )
        expect(testRenderer).toMatchSnapshot()
      })
    })
  })

  describe('when a theme is explicitly provided', () => {
    const theme = themes.natura.light

    it('should match snapshot - Intro with title and without detail', () => {
      testRenderer = TestRenderer.create(<Provider theme={theme}><Intro title="title" /></Provider>)

      expect(testRenderer).toMatchSnapshot()
    })

    it('should match snapshot - Intro with title and detail', () => {
      testRenderer = TestRenderer.create(<Provider theme={theme}><Intro title="title" detail="detail" /></Provider>)

      expect(testRenderer).toMatchSnapshot()
    })

    it('should match snapshot - Intro with title and icon', () => {
      testRenderer = TestRenderer.create(<Provider theme={theme}><Intro title={<TitleWithIcon />} /></Provider>)

      expect(testRenderer).toMatchSnapshot()
    })

    describe('when titleVariant property is undefined', () => {
      it('should match snapshot', () => {
        testRenderer = TestRenderer.create(
          <Provider theme={theme}>
            {/* eslint-disable-next-line no-undefined */}
            <Intro title="Test title" titleVariant={undefined as TypographyVariant} />
          </Provider>
        )
        expect(testRenderer).toMatchSnapshot()
      })
    })

    describe('when an invalid titleVariant property is provided', () => {
      it('should match snapshot', () => {
        testRenderer = TestRenderer.create(
          <Provider theme={theme}>
            <Intro title="Test title" titleVariant={'invalid' as TypographyVariant} />
          </Provider>
        )
        expect(testRenderer).toMatchSnapshot()
      })
    })

    Object.keys(variants).forEach((variant: string) => {
      it(`should match snapshot - Any valid Intro with ${variant} titleVariant`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Intro titleVariant={variant as TypographyVariant} title="title" /></Provider>)

        expect(testRenderer).toMatchSnapshot()
      })

      it(`should match snapshot - Any valid Intro with ${variant} detailVariant`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}>
          <Intro
            detail="detail"
            detailVariant={variant as TypographyVariant}
            title="title"
          />
        </Provider>)

        expect(testRenderer).toMatchSnapshot()
      })
    })

    Object.keys(colors).forEach((color: string) => {
      it(`should match snapshot - Any valid Intro with ${color} titleColor`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Intro titleColor={color as TypographyColor} title="title" /></Provider>)

        expect(testRenderer).toMatchSnapshot()
      })

      it(`should match snapshot - Any valid Intro with ${color} detailColor`, () => {
        testRenderer = TestRenderer.create(<Provider theme={theme}><Intro detail="detail" detailColor={color as TypographyColor} title="title" /></Provider>)

        expect(testRenderer).toMatchSnapshot()
      })
    })
  })
})
