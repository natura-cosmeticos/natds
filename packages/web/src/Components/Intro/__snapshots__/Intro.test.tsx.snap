// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with body1 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body1 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with body1 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-body1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with body2 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with body2 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-body2 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with button detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-button MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with button titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-button MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with caption detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-caption MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with caption titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-caption MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with error detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorError"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with error titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorError"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h1 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h1 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h1 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h2 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h2 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h2 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h3 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h3 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h3 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h3 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h4 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h4 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h4 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h4 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h5 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h5 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h5 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h5 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h6 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-h6 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with h6 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-h6 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with inherit detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorInherit"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with inherit detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with inherit titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorInherit"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with inherit titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with initial detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with initial titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with overline detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-overline MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with overline titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-overline MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with primary detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with primary titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with secondary detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorSecondary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with secondary titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorSecondary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with subtitle1 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with subtitle1 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with subtitle2 detailVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with subtitle2 titleVariant 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with textPrimary detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with textPrimary titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with textSecondary detailColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorTextSecondary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Any valid Intro with textSecondary titleColor 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Intro with title and detail 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
  <p
    className="MuiTypography-root NatDSIntro-introDetail-2 MuiTypography-body2 MuiTypography-colorTextPrimary"
  >
    detail
  </p>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Intro with title and icon 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    <i
      aria-hidden={true}
      className="material-icons MuiIcon-root natds-icons natds-icons-filled-action-add NatDSIcon-root-5 NatDSIcon-root-6"
    />
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided should match snapshot - Intro with title and without detail 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided when an invalid titleVariant property is provided should match snapshot 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-colorTextPrimary"
  >
    Test title
  </h3>
</section>
`;

exports[`Intro component when a theme is explicitly provided when titleVariant property is undefined should match snapshot 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    Test title
  </h3>
</section>
`;

exports[`Intro component when no theme is explicitly provided and titleVariant property is undefined should match snapshot 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-4 MuiTypography-subtitle1 MuiTypography-colorTextPrimary"
  >
    Test title
  </h3>
</section>
`;

exports[`Intro component when no theme is explicitly provided when an invalid titleVariant property is provided should match snapshot 1`] = `
<section
  className="NatDSIntro-root-1"
>
  <h3
    className="MuiTypography-root NatDSIntro-introTitle-3 NatDSIntro-introTitle-5 MuiTypography-colorTextPrimary"
  >
    Test title
  </h3>
</section>
`;
